File: app/src/main/java/ru/buylist/utils/BuylistApp.java
package ru.buylist.utils;

import android.app.Application;

import ru.buylist.data.DataRepository;
import ru.buylist.data.TemporaryDataStorage;
import ru.buylist.data.db.BuyListDatabase;

public class BuylistApp extends Application {
    private static BuylistApp instance;
    private AppExecutors executors;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        executors = new AppExecutors();
    }

    public static BuylistApp instance() {
        return instance;
    }

    public BuyListDatabase getDatabase() {
        return BuyListDatabase.getInstance(this, executors);
    }

    public DataRepository getRepository() {
        return DataRepository.get(getDatabase(), executors);
    }

    public TemporaryDataStorage getStorage() {
        return TemporaryDataStorage.instance(this);
    }
}

File: app/src/main/java/ru/buylist/BuyListApp.kt
package ru.buylist

import android.app.Application
import ru.buylist.data.db.BuyListDatabase

class BuyListApp : Application() {

    override fun onCreate() {
        super.onCreate()
        INSTANCE = this
    }

    fun getDatabase(): BuyListDatabase = BuyListDatabase.getInstance(INSTANCE)

    companion object {
        private lateinit var INSTANCE : BuyListApp

        @JvmStatic
        fun get() : BuyListApp = INSTANCE
    }
}